Assets/Scripts/EnemyScripts/BossBullet.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BossBullet : MonoBehaviour
+{
+    public GameObject player;
+    private Rigidbody2D rb;
+    public float bulletSpeed;
+    private float range;
+    private Vector3 startPos;
+
+    // Variable to hold the bullet damage
+    private int bulletDamage;
+
+    void Start()
+    {
+        startPos = transform.position;
+        rb = GetComponent<Rigidbody2D>();
+        player = GameObject.FindGameObjectWithTag("Player");
+
+        // Set bullet damage based on current wave
+        bulletDamage = 30 + (GameSettings.waveNumber / 5) * 5; // Increase damage by 5 for every 5 levels
```

Comment:
you could maybe serialize the variables and use them here instead of hard coding the numbers e.g. 
``
bulletDamage + (GameSettings.waveNumber / 5) * 5; 
``
instead of
``
30 + (GameSettings.waveNumber / 5) * 5;
``
This would make it easier to tweak the numbers in the future.